app/code/Magento/Catalog/Console/Command/ImagesResizeCommand.php

```diff
@@ -79,7 +79,8 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $productIds = $productCollection->getAllIds();
         if (!count($productIds)) {
             $output->writeln("<info>No product images to resize</info>");
-            return;
+            // we must have an exit code higher than zero to indicate something was wrong
+            return \Magento\Framework\Console\Cli::RETURN_FAILURE;
```

Comment:
@LeeSaferite I rather would have it as an error so you can choose to ignore it in your 'external' tooling